src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java

```diff
@@ -87,22 +98,14 @@ private void generateImage(ImageParam imageParam) throws IOException {
      * @throws IOException when is a invalid file.
      */
     private void generateThumbnail(ImageParam imageParam) throws IOException {
-        commonCommandImage(imageParam);
-
-        commandStrategyList.add(new CommandThumbnail(imageParam.isThumbnail()));
-        commandStrategyList.add(new CommandOutputFilePath(imageParam.getOutputPathFile()));
-        commandStrategyList.add(new CommandOutputFileName(imageParam.getOutputFileName()));
-        commandStrategyList.add(new CommandImageFormat(imageParam.getImageFormat()));
-    }
-
-    /**
-     * Generates a command common.
-     * @param imageParam receives image params.
-     * @throws IOException when is a invalid file.
-     */
-    private void commonCommandImage(ImageParam imageParam) throws IOException {
-        commandStrategyList.add(new CommandImageMagickPath());
-        commandStrategyList.add(new CommandImageConverter());
-        commandStrategyList.add(new CommandInputFilePath(imageParam.getInputPathFile()));
+        final String THUMBNAIL_TAG = "thumb01";
+
+        commandThumbnailList.add(new CommandImageMagickPath());
+        commandThumbnailList.add(new CommandImageConverter());
+        commandThumbnailList.add(new CommandInputFilePath(imageParam.getInputPathFile()));
+        commandThumbnailList.add(new CommandThumbnail(imageParam.isThumbnail()));
+        commandThumbnailList.add(new CommandOutputFilePath(imageParam.getOutputPathFile()));
+        commandThumbnailList.add(new CommandOutputFileName(THUMBNAIL_TAG));
+        commandThumbnailList.add(new CommandImageFormat(imageParam.getImageFormat()));
```

Comment:
I thing thats too, be methods

Reply:
They are in a private method "generateThumbnail" too

Reply:
Yes, but a method for all lines add to the list